### DISCUSSION TITLE
Copilot Chat connection errors

### DISCUSSION THREAD

**Discussion by decimvs** [community/community#121636] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

I'm experimenting some connection errors in the Copilot Chat. When I start a new session, it tries to get a token ant it works properly, but when it sends the request to the server no response is sent back. It tries ten times after it cancels the session. In the logs window a connection error is logged, but no more information is given.
The code suggestion in Visual Studio works properly, when I write some code it suggests me some code. When I see the logs window for code suggestion, calls to the API works correctly.

We use a Cisco Umbrella solution in our network, so it's highly possible that the connection is blocked. Can someone tell me what URL is calling Copilot Chat?

**Comment by Akash1134** [comment, marked as answer]
Hi @decimvs ,

Copilot communicates with GitHub servers for retrieving suggestions and performing its functionalities. However, the exact URLs it uses for its internal operations are not publicly disclosed for security reasons. In a corporate environment with a firewall or network restrictions, you should ensure that all GitHub domains are whitelisted which would include github.com and all its subdomains. Refer this [documentation](https://docs.github.com/en/copilot/troubleshooting-github-copilot/troubleshooting-firewall-settings-for-github-copilot) for troubleshooting firewall settings and [this space](https://docs.github.com/en/copilot/troubleshooting-github-copilot/troubleshooting-network-errors-for-github-copilot) for other network related issues. 

Please note, if the issue persists, you should [reach out directly to GitHub support](https://support.github.com/) or your network administrator for a more tailored assistance.
